Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java

```diff
@@ -578,79 +574,96 @@ public UpdateResult updateFields(byte[] originalJsonObject, JsonNode jsonToUpdat
 		UpdateResult updateResult = new UpdateResult(jsonToUpdate, resultJson);
 		JsonNode node = objectMapper.readTree(new String(originalJsonObject));
 		ObjectNode objectNode = (ObjectNode) node;
-		if (attrId != null) { // partial update, remove original instanceid and update temporal properties
-			// remove instanceId from original json, if exists
+		if (attrId != null) {
 			if (objectNode.get(attrId) == null) {
 				throw new ResponseException(ErrorType.NotFound, "Provided attribute is not present");
 			}
 			JsonNode originalNode = ((ArrayNode) objectNode.get(attrId)).get(0);
 			if (((ObjectNode) originalNode).has(NGSIConstants.NGSI_LD_INSTANCE_ID)) {
 				((ObjectNode) originalNode).remove(NGSIConstants.NGSI_LD_INSTANCE_ID);
 			}
-			// keep original createdAt value if present in the original json
-			String createdAt = now;
-			if (((ObjectNode) originalNode).has(NGSIConstants.NGSI_LD_CREATED_AT)
-					&& ((ObjectNode) originalNode).get(NGSIConstants.NGSI_LD_CREATED_AT).isArray()) {
-				createdAt = ((ObjectNode) ((ObjectNode) originalNode).get(NGSIConstants.NGSI_LD_CREATED_AT).get(0))
-						.get(NGSIConstants.JSON_LD_VALUE).asText();
+			Iterator<String> it = jsonToUpdate.fieldNames();
```

Comment:
please remove this iterator, this is no where used.

Reply:
Removed